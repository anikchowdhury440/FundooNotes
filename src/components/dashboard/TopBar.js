import React, {Component} from 'react';
import {View, TouchableOpacity} from 'react-native';
import {Appbar, Avatar} from 'react-native-paper';
import { Strings } from '../../Language/Strings';
import TopBarStyle from '../../styles/TopBar.style';

export default class TopBar extends Component {
    constructor(props) {
        super(props);
    }

    handleMenuButton = async () => {
        const {onPress} = this.props
        this.props.navigation.openDrawer();
        //onPress();
    }

    render() {
        return (
            <View>
                <Appbar style = {TopBarStyle.container}>
                    <Appbar.Action
                        icon = 'menu'
                        onPress = {this.handleMenuButton}
                        />
                    <Appbar.Content
                        titleStyle = {TopBarStyle.appbar_content_style}
                        title = {Strings.searchYourNotes}
                        />
                    <Appbar.Action
                        style = {{marginRight : 10}}
                        icon = {(this.props.listView) ? 'view-grid-outline' : 'view-agenda-outline'}
                        onPress={this.props.onPressView}
                        />
                    <TouchableOpacity
                        style = {TopBarStyle.avatar_style}
                        onPress = {this.props.onPressProfile}>
                        <Avatar.Image size={24} 
                            source = {require('../../assets/blank-profile.png')} />
                    </TouchableOpacity>
                </Appbar>
            </View>
        )
    }
}