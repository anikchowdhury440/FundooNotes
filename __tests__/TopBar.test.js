import React from 'react';
import {shallow} from 'enzyme';
import TopBar from '../src/components/dashboard/TopBar';

describe('test TopBar', () => {
    it('should match to snapshot', () => {
        const component = shallow(<TopBar/>)
        expect(component).toMatchSnapshot();
    })

    it('test onPress event of list view icon or grid view icon it will change the listView State', async () => {
        const onPressEvent = jest.fn();
        const component = shallow(<TopBar onPress = {onPressEvent}/>)
        const instance = component.instance();
        expect(instance.state.listView).toBe(true);
        await instance.selectView();
        expect(onPressEvent).toHaveBeenCalled();
        expect(onPressEvent).toHaveBeenCalledTimes(1);
        expect(instance.state.listView).toBe(false);
        await instance.selectView();
        expect(onPressEvent).toHaveBeenCalled();
        expect(onPressEvent).toHaveBeenCalledTimes(2);
        expect(instance.state.listView).toBe(true);
    })
})