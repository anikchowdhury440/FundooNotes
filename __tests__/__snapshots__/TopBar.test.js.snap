// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TopBar should match to snapshot 1`] = `ShallowWrapper {}`;
